coordinate.go

```diff
@@ -22,15 +22,16 @@ const (
 	MetaSegmentKey = "consul-network-segment"
 )
 
-var (
-	// The maximum time to wait for a ping to complete.
-	MaxRTT = 5 * time.Second
-)
+type nodeChannel <-chan []*api.Node
+
+// The maximum time to wait for a ping to complete.
+var MaxRTT = 5 * time.Second
 
 // updateCoords is a long running goroutine that attempts to ping all external nodes
 // once per CoordinateUpdateInterval and update their statuses in Consul.
-func (a *Agent) updateCoords(nodeCh <-chan []*api.Node) {
+func (a *Agent) updateCoords(nodeCh nodeChannel) {
 	// Wait for the first node ordering
+	nodeCh = a.checkNodeTracking(nodeCh)
```

Comment:
This is a nice way to introduce a fix without modifying too much of the existing code. Should we be concerned about the complexity of introducing another channel and goroutine into the system, however? From my understanding, it appears as if we could do the following, but perhaps I'm missing something:

```
func (a *Agent) checkNodeTracking(nodes []*api.Node) {...}
...
select {
case newNodes := <-nodeCh:
        a.checkNodeTracking(newNodes)
	if len(newNodes) != len(nodes) {
```

Also, maybe something that implies a state change is occurring would be more understandable for the method name. Something like `clearDeregisteredNodeTracking()` might be easier for other authors.

Reply:
I originally had a call almost just like that but given the architecture of ESM I thought I'd start with patterns more inspired by CSP and data flow programming and focus on the channels as a primary interaction point. We'll see how it goes.